src/libcore/ptr/mod.rs

```diff
@@ -1929,18 +1934,23 @@ impl<T: ?Sized> *mut T {
     /// of the returned pointer, nor can it ensure that the lifetime `'a`
     /// returned is indeed a valid lifetime for the contained data.
     ///
-    /// When calling this method, you have to ensure that if the pointer is
-    /// non-NULL, then it is properly aligned, dereferencable (for the whole
-    /// size of `T`) and points to an initialized instance of `T`. This applies
-    /// even if the result of this method is unused!
+    /// When calling this method, you have to ensure that *either* the pointer is NULL *or*
+    /// all of the following is true:
+    /// - it is properly aligned
+    /// - it must point to an initialized instance of T; in particular, the pointer must be
+    ///   "dereferencable" in the sense defined [here].
+    ///
+    /// This applies even if the result of this method is unused!
     /// (The part about being initialized is not yet fully decided, but until
     /// it is the only safe approach is to ensure that they are indeed initialized.)
     ///
     /// Additionally, the lifetime `'a` returned is arbitrarily chosen and does
-    /// not necessarily reflect the actual lifetime of the data. It is up to the
-    /// caller to ensure that for the duration of this lifetime, the memory this
-    /// pointer points to does not get accessed through any other pointer.
+    /// not necessarily reflect the actual lifetime of the data. Moreover, you *must* enforce
```

Comment:
```suggestion
    /// not necessarily reflect the actual lifetime of the data. *You* must enforce
```
This is my last nit. ;)

Reply:
I'm not a native English speaker (surely you can tell that), language tweaks are more than welcome :)

Reply:
I'm not a native speaker either, but doing a PhD supervised by a native speaker helps to increase one's writing skills. ;)

Reply:
Btw, when there are multiple "suggestions" on GH, you can go do the diff view and accept them all at once, avoiding dozens of tiny commits.